src/input.rs

```diff
@@ -475,23 +499,34 @@ impl<'a, A: ActionContext + 'a> Processor<'a, A> {
         }
     }
 
-    fn scroll_terminal(&mut self, code: u8, modifiers: ModifiersState) {
+    fn scroll_terminal(&mut self, code: u8, modifiers: ModifiersState, scroll_multiplier: u8) {
         debug_assert!(code == 64 || code == 65);
 
-        let faux_scrollback_lines = self.mouse_config.faux_scrollback_lines;
         let mouse_modes = TermMode::MOUSE_REPORT_CLICK | TermMode::MOUSE_DRAG | TermMode::MOUSE_MOTION;
+
+        // Make sure the new and deprecated setting are both allowed
+        let faux_scrollback_lines = self.mouse_config
+            .faux_scrollback_lines
+            .unwrap_or(self.scrolling_config.faux_multiplier as usize);
+
         if self.ctx.terminal_mode().intersects(mouse_modes) {
             self.mouse_report(code, ElementState::Pressed, modifiers);
-        } else if faux_scrollback_lines > 0 {
+        } else if self.ctx.terminal_mode().contains(TermMode::ALT_SCREEN)
+            && faux_scrollback_lines > 0 && !modifiers.shift
+        {
             // Faux scrolling
             let cmd = code + 1; // 64 + 1 = A, 65 + 1 = B
-            let mut content = Vec::with_capacity(faux_scrollback_lines * 3);
+            let mut content = Vec::with_capacity(faux_scrollback_lines as usize * 3);
             for _ in 0..faux_scrollback_lines {
                 content.push(0x1b);
                 content.push(b'O');
                 content.push(cmd);
             }
             self.ctx.write_to_pty(content);
+        } else {
+            for _ in 0..scroll_multiplier {
+                self.ctx.scroll(Scroll::Lines(-((code as isize) * 2 - 129)));
```

Comment:
This seems like a lot of vodoo magic with the `- 129`. Might be good to document this?